File: 0243. Shortest Word Distance/main0243.py



class Solution243(object):
    def shortestDistance(self, wordsDict, word1, word2):
        """
        :type wordsDict: List[str]
        :type word1: str
        :type word2: str
        :rtype: int
        """
        i1, i2 = -1, -1
        res = len(wordsDict) + 1
        for x in range(len(wordsDict)):
            if wordsDict[x] == word1:
                i1 = x
            elif wordsDict[x] == word2:
                i2 = x
            if i1 != -1 and i2 != -1:
                res = min(res, abs(i2 - i1))
                
        return res

File: 0243. Shortest Word Distance/main0243.java
public class main0243 {
    public static void main(String[] args) {
        
    }
}


class Solution0243 {
    public int shortestDistance(String[] wordsDict, String word1, String word2) {
        int res = wordsDict.length;
        int prev = -1; // meet prev index of target 2
        for (int i = 0; i < wordsDict.length; i++) {
            if (wordsDict[i].equals(word1)) {
                if (prev != -1 && wordsDict[prev].equals(word2)) {
                    res = Math.min(i - prev, res);
                }
                prev = i;
            } else if (wordsDict[i].equals(word2)) {
                if (prev != -1 && wordsDict[prev].equals(word1)) {
                    res = Math.min(i - prev, res);
                }
                prev = i;
            }
        }
        return res;
    }
}